src/inmanta/server/services/environmentservice.py

```diff
@@ -601,3 +601,18 @@ async def notify_listeners(
                     await listener.environment_action_updated(updated_env, original_env)
             except Exception:
                 LOGGER.warning(f"Notifying listener of {action} failed with the following exception", exc_info=True)
+
+    async def register_setting(self, env: data.Environment, setting: Setting) -> None:
+        """
+        Should only be called during pre-start
+        Adds a new setting to this environment from outside inmanta-core.
+        As example, inmanta-lsm can use this methode to add settings that are only
+        relevant for inmanta-lsm but that are needed in the environment.
+
+        :param env: the environment in which the setting will be registered
+        :param setting: the setting that should be added to the existing settings
+        """
+        try:
+            await env.register_setting(setting)
+        except KeyError:
+            raise
```

Comment:
This except-clause is redundant.
```suggestion
        await env.register_setting(setting)
```